src/lib/api/org/index.ts

```diff
@@ -275,3 +275,85 @@ function convertToSnykProject(projectData: v3ProjectData[]) : SnykProject[] {
   return projects
 }
 
+export async function listTargets(
+  requestManager: requestsManager,
+  orgId: string,
+): Promise<TargetsResponse> {
+  getApiToken();
+  getSnykHost();
+  debug(`Listing all targets for org: ${orgId}`);
+
+  if (!orgId) {
+    throw new Error(
+      `Missing required parameters. Please ensure you have set: orgId, settings.
+        \nFor more information see: https://apidocs.snyk.io/?version=2022-09-15%7Ebeta#get-/orgs/-org_id-/targets`,
+    );
+  }
+
+  const targets = await listAllSnykTargetREST(requestManager, orgId)
+  
+  const snykTargetData: TargetsResponse = {
+    targets: targets
+  }
+
+  return snykTargetData;
+}
+
+export async function listAllSnykTargetREST(requestManager: requestsManager,
+  orgId: string,
+  ): Promise<SnykTargetREST[]> {
+
+    let lastPage = false;
+    let SnykTargetRESTList: SnykTargetREST[] = [];
+    let pageCount = 1;
+    let nextPageLink: string | undefined = undefined;
+    while (!lastPage) {
```

Comment:
I see this pagination logic in many places, shouldn't we have one main script that works with our REST API and handles pagination?

Reply:
like, discuss the other day, not sure how to do this now. Will have a look. I don't think this should block us.

Reply:
See https://snyksec.atlassian.net/jira/software/c/projects/TECHSERV/boards/387/backlog?view=detail&selectedIssue=TECHSERV-3387&issueLimit=100